app/src/main/java/com/github/sdpsharelook/camera/CameraActivity.kt

```diff
@@ -0,0 +1,111 @@
+package com.github.sdpsharelook.camera
+
+import android.Manifest
+import android.app.Activity
+import android.app.AlertDialog
+import android.content.ContentValues
+import android.content.Context
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.net.Uri
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.os.Environment
+import android.provider.MediaStore
+import android.view.View
+import android.widget.Button
+import android.widget.ImageView
+import android.widget.Toast
+import androidx.activity.result.ActivityResultCallback
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.core.app.ActivityCompat
+import androidx.core.content.ContextCompat
+import androidx.core.content.FileProvider
+import com.github.sdpsharelook.R
+import com.github.sdpsharelook.Utils
+import java.io.File
+import java.text.SimpleDateFormat
+import java.util.*
+
+class CameraActivity : AppCompatActivity() {
+
+    private var currentPath: String? = null
+    private var hasPermissions = false
+
+    private fun showAlert(message: String) {
+        val builder = AlertDialog.Builder(this as Context)
+        builder.setMessage(message)
+        builder.setPositiveButton(R.string.button, null)
+        val dialog = builder.create()
+        dialog.show()
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_camera)
+        findViewById<Button>(R.id.buttonTakePic).setOnClickListener() {
+            takePic()
+        }
+    }
+
+    fun takePic() {
+        tempImageUri = FileProvider.getUriForFile(this, "camera", createImage())
+        checkPermissions(Manifest.permission.CAMERA)
+        if (checkPerms()) {
+            cameraLauncher.launch(tempImageUri)
+        }
+    }
+
+    private var tempImageUri: Uri? = null
+    private val cameraLauncher = registerForActivityResult(ActivityResultContracts.TakePicture()) { success ->
+        if (success) {
+            val file = File(currentPath)
```

Comment:
Maybe we should clearly define later what goes to the main storage and what feeds the cache. I think not all pictures should be saved for lifetime.